src/System.Management/tests/System/Management/ManagementObjectTests.cs

```diff
@@ -50,5 +49,31 @@ public void Set_Property_Win32_ComputerSystem()
                 obj.SetPropertyValue("Workgroup", "WmiTests");
             }
         }
+
+        [ConditionalFact(typeof(PlatformDetection), nameof(PlatformDetection.IsNotWindowsNanoServer))]
+        [OuterLoop]
+        public void Invoke_Instance_And_Static_Method_Win32_Process()
+        {
+            var processClass = new ManagementClass("Win32_Process");
+            object[] methodArgs = { "notepad.exe", null, null, 0 };
+
+            object resultObj = processClass.InvokeMethod("Create", methodArgs);
+
+            Thread.Sleep(1000);
```

Comment:
Can we please avoid sleeps? If a delay is unavoidable., check, if false, sleep(100), check again, up to 10 times say, then delay may be between 0-1000 ms.

Reply:
Better though to get the PID (apparently methodArgs[3]) and use our Process class to WaitforInputIdle so you don't wait a moment more than necessary.

But what is the sleep for -- is it not possible to get the PID for 1 second? seems odd.

Reply:
Yes, I don't like those either. we probably need to use some other stuff to ensure that the new process is up and running and later stopped.

Reply:
Even better than that after reading the docs about these wmi methods we actually don't need any wait: the methods are synchronous.